csfieldguide/templates/interactives/mips-assembler.html

```diff
@@ -0,0 +1,186 @@
+            </label>
+          </div>
+          <div class="col">
+            <button id="reset-adv" type="button" class="btn btn-outline-danger">{% trans 'Load advanced example' %}</button>
+            <button id="reset-basic" type="button" class="btn btn-outline-danger">{% trans 'Load basic example' %}</button>
+          </div>
+          <div class="col">
+            <button id="submit-mips" type="button" class="btn btn-success">{% trans 'Assemble!' %}</button>
+          </div>
+        </div>
+      </div>
+
+      <!-- Output Column -->
+      <div class="col">
+        <!-- Initially a storage of the advanced example program, it is immediately cleared -->
+        <div id="assembler-output" class="code scroll h-100">
+            {% trans '# Define the data strings' %}
+            .data
+            go_str:   .asciiz "{% trans 'GO!!!!!' %}\n"
+            new_line: .asciiz "\n"
+            
+            .text
+            {% trans '# Where should we start?' %}
+            .globl main
+            
+            main:
+              {% trans '# Put our starting value 5 into register $t0. We will update it as we go' %}
+              li $t0, 5
+              {% trans '# Put our stopping value 0 into register $t1' %}
+              li $t1, 0
+            
+            {% trans '# This label is just used for the jumps to refer to' %}
+            start_loop:
+              {% blocktrans %}# This says that if the values in $t0 and $t1 are the same,
+              # it should jump down to the end_loop label. This is the
+              # main loop condition{% endblocktrans %}
+              beq $t0, $t1, end_loop
+              {% blocktrans %}# These three lines prepare for and print the current int
+              # It must be moved into $a0 for the printing{% endblocktrans %}
+              move $a0, $t0
+              li $v0, 1
+              syscall
+              {% blocktrans %}# These three lines print a new line character so that
+              # each number is on a new line{% endblocktrans %}
+              li $v0, 4
+              la $a0, new_line
+              syscall
+              {% trans '# Add -1 to the value in $t0, i.e decrement it by 1' %}
+              addi $t0, $t0, -1
+              {% trans '# Jump back up to the start_loop label' %}
+              j start_loop
+            
+            {% blocktrans %}# This is the end loop label that we jump to
+            # when the loop condition becomes true{% endblocktrans %}
+            end_loop:
+              {% trans '# These three lines print the “GO!!!!” string.' %}
+              li $v0, 4
+              la $a0, go_str
+              syscall
+              {% trans '# And these two lines make the program exit nicely' %}
+              li $v0, 0
+              jr $ra
+        </div>
+      </div>
+
+    </div>
+
+    <!-- Contributions -->
+    <div class="row justify-content-center">
```

Comment:
Consider adding some padding to the top of thise div so there is some space between the interactive and the contribution. Maybe pt-5.